addons/resource/models/resource.py

```diff
@@ -689,6 +691,33 @@ def _onchange_hours(self):
         # avoid wrong order
         self.hour_to = max(self.hour_to, self.hour_from)
 
+    def _compute_this_week(self):
+        this_week_type = self.get_week_type(fields.Date.context_today(self))
+        for attendance in self:
+            attendance.is_this_week = this_week_type == int(attendance.week_type)
+
+    @api.model
+    def get_week_type(self, date):
+        # week_type is defined by
+        #  * counting the number of days from January 1 of year 1
+        #    (extrapolated to dates prior to the first adoption of the Gregorian calendar)
+        #  * converted to week numbers and then the parity of this number is asserted.
+        # It ensures that an even week number always follows an odd week number. With classical week number,
+        # some years have 53 weeks. Therefore, two consecutive odd week number follow each other (53 --> 1).
+        return int(math.floor((date.toordinal() - 1) / 7) % 2)
+
+    def _compute_display_name(self):
+        this_week_type = str(self.get_week_type(fields.Date.context_today(self)))
+        section_names = {'0': _('First week'), '1': _('Second week')}
+        section_info = {True: _('this week'), False: _('other week')}
+        for record in self:
+            if record.display_type == 'line_section':
+                section_name = "%s (%s)" % (section_names[record.week_type], section_info[this_week_type == record.week_type])
+                record.display_name = section_name
+                record.name = section_name
```

Comment:
It's useful somewhere? I'm not sure that is a good idea, as each time we call `_compute_display_name` we will write on this stored field. And if we don't call this function for a while, the value `name` will not be correct.